### DISCUSSION TITLE
Can i add new authentication on live production mode?

### DISCUSSION THREAD

**Discussion by satzofficial** [opcodesio/log-viewer#344] [Ideas]
Hi
I want to implement new authentication for specific users at this page only. like this 
![image](https://github.com/opcodesio/log-viewer/assets/140888681/4e4a5bf6-cad5-47e0-bedd-699ef7f66df3)

**Comment by arukompas** [comment]
hey @satzofficial 

You can create your own custom Middleware which uses vanilla PHP's [HTTP Basic auth](https://www.php.net/manual/en/features.http-auth.php) to get the username/password via a form like you showed. This way you control whether to let the user through or not. 

But keep in mind that you're in control of this authentication, meaning you have to store the passwords somewhere (e.g. `.env` file or database) and compare them yourself.

Once you have the middleware ready, just add it to the `config/log-viewer.php` in the `middleware` and `api_middleware` arrays.
